.github/workflows/cypress-workflow.yml

```diff
@@ -0,0 +1,129 @@
+          wget https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-${{ steps.kibana_version.outputs.kibana_version }}-darwin-x86_64.tar.gz
+          tar -xzf elasticsearch-oss-${{ steps.kibana_version.outputs.kibana_version }}-darwin-x86_64.tar.gz
+          cd elasticsearch-${{ steps.kibana_version.outputs.kibana_version }}/
+          bin/elasticsearch-plugin install -b https://d3g5vo6xdbdb9a.cloudfront.net/downloads/elasticsearch-plugins/opendistro-security/opendistro_security-${{ steps.opendistro_version.outputs.opendistro_version }}.zip
+          bash plugins/opendistro_security/tools/install_demo_configuration.sh -y -i
+          bin/elasticsearch &
+          sleep 60
+          curl https://localhost:9200 -u admin:admin -k
+      - name: Checkout Kibana
+        uses: actions/checkout@v2
+        with:
+          repository: opendistro-for-elasticsearch/kibana-oss
+          ref: ${{ steps.kibana_version.outputs.kibana_version }}
+          token: ${{ secrets.KIBANA_OSS_TOKEN }}
+          path: kibana
+      - name: Get node and yarn versions
+        id: versions
+        run: |
+          echo "::set-output name=node_version::$(node -p "(require('./kibana/package.json').engines.node).match(/[.0-9]+/)[0]")"
+          echo "::set-output name=yarn_version::$(node -p "(require('./kibana/package.json').engines.yarn).match(/[.0-9]+/)[0]")"
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ steps.versions.outputs.node_version }}
+          registry-url: 'https://registry.npmjs.org'
+      - name: Install correct yarn version for Kibana
+        run: |
+          npm uninstall -g yarn
+          echo "Installing yarn ${{ steps.versions_step.outputs.yarn_version }}"
+          npm i -g yarn@${{ steps.versions.outputs.yarn_version }}
+      - name: Bootstrap plugin/kibana
+        env:
+          CYPRESS_INSTALL_BINARY: ${{ steps.cypress_version.outputs.cypress_version }}
+        run: |
+          mkdir -p kibana/plugins
+          mv security-kibana-plugin kibana/plugins
+          cd kibana/plugins/security-kibana-plugin
+          yarn kbn bootstrap
+      - name: Run kibana server
+        run: |
+          security_config='elasticsearch.username: "kibanaserver"
+          elasticsearch.password: "kibanaserver"
+          elasticsearch.hosts: "https://localhost:9200"
+          elasticsearch.ssl.verificationMode: none
+          elasticsearch.requestHeadersWhitelist: ["securitytenant","Authorization"]
+          opendistro_security.multitenancy.enabled: true
+          opendistro_security.multitenancy.tenants.preferred: ["Private", "Global"]
+          opendistro_security.readonly_mode.roles: ["kibana_read_only"]
+          opendistro_security.auth.type: "basicauth"
+          '
+          echo "$security_config" >> kibana/config/kibana.yml
+          cat kibana/config/kibana.yml
+          cd kibana
+          yarn start --no-base-path --no-watch &
```

Comment:
do we need to get the PID and kill the background processes at the end of the workflow?

Reply:
Good question. I checked the Github workflow documentation. 
https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners

"Each job in a workflow executes in a fresh instance of the virtual machine." So, the cross job interaction is not a concern. Also I would assume that it will take care of the teardown.